src/controller/p2p/preheat/controller.go

```diff
@@ -2,15 +2,23 @@ package preheat
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/orm"
 	"github.com/goharbor/harbor/src/lib/q"
 	"github.com/goharbor/harbor/src/pkg/p2p/preheat/instance"
 	policyModels "github.com/goharbor/harbor/src/pkg/p2p/preheat/models/policy"
 	providerModels "github.com/goharbor/harbor/src/pkg/p2p/preheat/models/provider"
 	"github.com/goharbor/harbor/src/pkg/p2p/preheat/policy"
 	"github.com/goharbor/harbor/src/pkg/p2p/preheat/provider"
+	"github.com/goharbor/harbor/src/pkg/scheduler"
+)
+
+const (
+	// SchedulerCallback ...
+	SchedulerCallback = "P2PPreheatCallback"
```

Comment:
I did not find the callback func definition.

You should implement a callback func and register it with a name like `P2PPreheatCallback` to the scheduler.

In the callback func, you need to enforce the corresponding policy when the schedule signal is coming.

Reply:
implemented P2PPreheatCallback func